### DISCUSSION TITLE
Print time increase x2 [Chitubox:1.9.3,Phrozen:mini8k]

### DISCUSSION THREAD

**Discussion by bartoszek** [sn4k3/UVtools#597] [Q&A] [answered]
## Problem:
Print time showed in the `UVtools` is x2 longer than what the `Chitubox` shows.
From my experience with the `Phrozen mini 8k` actual print time is ~1h longer
than estimated by the `Chitubox` but not x2 longer.
I haven't alter any `print parameters` in `UVTools` just simple `auto repair`
with some resin trap filled and raft relief.
Printer seems to actually print slower, haven't measure it thought
but printer appears to wait some extra time before exposing the layer.

## Environment:
* Chitubox: 1.9.3
* UVTools: 3.8.0
* Printer: Phrozen mini 8k
![image](https://user-images.githubusercontent.com/1213348/199287210-012c7b21-5236-4087-b94f-695defac4fd3.png)
![image](https://user-images.githubusercontent.com/1213348/199289830-e1fd00fd-1e2b-49b2-9e4f-22a51eb5e709.png)

**Comment by sn4k3** [comment, marked as answer]
Print time is just an estimate, it can have a high or low error, depending on printer parameters and firmware.
Only printer can give you the real print time. None software can accurately give that information without known all variables that are hidden in the firmware and depend on each printer. Also softwares give instant time without take acceleration into account, and instant times are not what you get in real life.
Chitubox may give you the best print time because they own the firmwares and then they are able to perform more accurate calculations

**Reply by bartoszek** [reply]
Isn't `PrintTime` just a metadeta in the `ctb` file? which printer simply presents alongside the thumbnail on a touchscreen?
You're saying that the printer firmware calculate the actual `PrintTime`?
![image](https://user-images.githubusercontent.com/1213348/199322940-e96dc49f-81e6-4d1c-aab4-b5b0321529d5.png)

**Reply by sn4k3** [reply]
It is, but UVtools always recalculate print time based on modifications and also because some formats don't have that data. So it always recalculate print time on file open and on any change you made to layers

> You're saying that the printer firmware calculate the actual PrintTime?

Yes, when print is finished it show you the real print time. Before print is just another estimate with error.

**Reply by bartoszek** [reply]
If so shouldn't this value be kept if user doesn't alter the `printer settings`?

**Reply by sn4k3** [reply]
Most users have reported that PrintTime calculated by UVtools is the most accurate of the slicers, that way I have it to always recalculate no matter if file has the metadata.
If you have such high error that means you values aren't getting respected by printer and it's contraining to another values in the firmware, we can never know...
Still I can have a look into the calculation logic to see if theres a bug, if you want, send me your file and i will look into it

**Reply by bartoszek** [reply]
I'll try to make a synthetic example and post it here (before mentioned file is quite big ~500MB I doubt GH will allow it)

**Reply by sn4k3** [reply]
I need a file that you already print and know real print time.
You can use [wetransfer.com](https://wetransfer.com/) to upload the file

**Reply by bartoszek** [reply]
Example file ~250MB: https://drive.google.com/file/d/1ClQ_inXIOL9KL2hmXfdyLT53wWbkKIly/view?usp=share_link
* Chitubox time estimation: 10h 6m
* Phrozen Mini 8k print time: 11h 36m
* UVTools time extimation: 17h 58m
![image](https://user-images.githubusercontent.com/1213348/199745452-7a6d1c8d-d706-475d-ad09-70a18016a223.png)

**Reply by sn4k3** [reply]
Found the reason on your file.

Looks at this file properties:
```
PerLayerSettings = 15
RestTimeAfterLift = 0.3
RestTimeAfterRetract = 1
RestTimeAfterLift2 = 0.3
RestTimeAfterRetract2 = 1
RestTimeAfterLift3 = 0.3
RestTimeBeforeLift = 0.3
```

That means your rest times are: 1s/0.3s/0.3s

Now look at your layers data, all layers shows with: `WaitTimes: 6.47/0.3/0.3s`
Look back to file properties and find this `PerLayerSettings = 15`, a value of 15 or lower mean to not respect individual layer data but follow global parameters instead. So your printer is printing with 1s/0.3s/0.3s waits BUT UVtools uses layer data to compute time! So time is getting calculated with `6.47s*5146` instead of `1*5146`, that's where the time error comes!

If you reset wait times to make layer data to update it regulate the time as you expected giving a value near the Chitubox estimate:

(Left = Modified wait time)
(Right = Your original file)
![image](https://user-images.githubusercontent.com/113281/199841377-f6f31cd6-bfbc-4c32-a807-5a8af141ebf4.png)

Don't ask me why you got 6.47s on all your layer data, does it come with that value out of chitubox?

Anyway you should use and apply UVtools suggestions for the wait time, it will auto apply wait times correctly and also fix your print time by reset times.

**Reply by bartoszek** [reply]
Chitubox is a giant piece of crap imho, thanks for the tip.

**Reply by sn4k3** [reply]
I agree ;)
